Collaborator: CiCi Feng

# Was the randomization done correctly? (30pt)

In [1]:
# Load the necessary packages
import pandas as pd
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt
import statsmodels.api as sm
import statsmodels.formula.api as smf

    1. (10pt) Present your results in a single table with the following columns and 14 (or so) rows. The table should look something like this:
    Suggestion: use t-test to determine whether the difference between Tand Cvillages is statistically significant for each of the variables in the dataset. Focus only on the data from 1997 for poor. Ignore variables such as folnum and village that do not carry social significance. t-test can be done with t.test in R. For instance t.test(x, y) compares two unpaired vectors xand y, and outputs confidence intervals, p-value, and other things.
    Suggestion 2: There are many ways you can create this table, here is one suggestion you may follow:
    (a) create an empty data frame that contains the values you need: variable name, average for T, average for C, their difference, and p-value from t-test. In R, you can also just create a NULL-object:
    df <- NULL
    This will be your final data frame, the one you will print.
    (b) do a for-loop over all the variable names, data for which you want to compare. Inside the loop:
    (c) extract the variable (your loop variable) values for Tand Cgroup as separate vectors
    (d) compute averages of these two vectors
    (e) compute p-value for t-test between these two vectors
    (f) create a new one-line data frame (one row of the required table) that contains the values for this variable only.
    (g) attach this new one-line data frame to the final data frame (the one you created as empty). You can use rbind as
      df <- rbind(df, newDF)
      As a result, your data frame will contain all the required values for all the variables. You can use knitr::kable or xtable library to print it in a nicer way. You can use argument check.names=FALSE to create a data frame with non-syntactic variable names, e.g. ones that contain space like “p value”.

In [2]:
# Load the dataset
progresa_data = pd.read_csv('/home/jovyan/INFO371PS/Data/progresa-sample.csv', sep=',')
# Print basic info and first five lines
progresa_data.info()
progresa_data.head(5)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 77250 entries, 0 to 77249
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   year           77250 non-null  int64  
 1   sex            77226 non-null  float64
 2   indig          76950 non-null  float64
 3   dist_sec       77250 non-null  float64
 4   sc             68797 non-null  float64
 5   grc            70701 non-null  float64
 6   fam_n          77250 non-null  int64  
 7   min_dist       77250 non-null  float64
 8   dist_cap       77250 non-null  float64
 9   poor           77250 non-null  object 
 10  progresa       77250 non-null  object 
 11  hohedu         77250 non-null  int64  
 12  hohwag         77250 non-null  float64
 13  welfare_index  77040 non-null  float64
 14  hohsex         77230 non-null  float64
 15  hohage         77240 non-null  float64
 16  age            77250 non-null  int64  
 17  village        77250 non-null  int64  
 18  folnum

year  sex  indig  dist_sec   sc  grc  fam_n   min_dist   dist_cap   poor  \
0    97  0.0    0.0     4.473  1.0  7.0      7  21.168384  21.168384  pobre   
1    98  0.0    0.0     4.473  1.0  8.0      7  21.168384  21.168384  pobre   
2    97  1.0    0.0     4.473  1.0  6.0      7  21.168384  21.168384  pobre   
3    98  1.0    0.0     4.473  1.0  7.0      7  21.168384  21.168384  pobre   
4    97  0.0    0.0     4.473  1.0  2.0      7  21.168384  21.168384  pobre   

   ... hohedu  hohwag  welfare_index  hohsex  hohage  age  village  folnum  \
0  ...      6     0.0          583.0     1.0    35.0   13      163       1   
1  ...      6     0.0          583.0     1.0    35.0   14      163       1   
2  ...      6     0.0          583.0     1.0    35.0   12      163       2   
3  ...      6     0.0          583.0     1.0    35.0   13      163       2   
4  ...      6     0.0          583.0     1.0    35.0    8      163       3   

   grc97  sc97  
0      7   1.0  
1      7   1.0  
2      6   1.0  
3      6   1.0  
4      2   1.0  

[5 rows x 21 columns]

In [3]:
# Filtering the data
control_97 = progresa_data[(progresa_data.year == 97) & (progresa_data.poor == "pobre") & (progresa_data.progresa == "0")]
treatment_97 = progresa_data[(progresa_data.year == 97) & (progresa_data.poor == "pobre") & (progresa_data.progresa == "basal")]
# Create an empty list to store results
output = []
# For loop to calculate control and treatment group average, calculate difference, calculate p value
for column in progresa_data.columns:
    if column not in ['year', 'poor', 'progresa', 'village', 'folnum', 'grc97', 'sc97']:
        control_avg = np.mean(control_97[column])
        treatment_avg = np.mean(treatment_97[column])
        p_value = stats.ttest_ind(treatment_97[~treatment_97[column].isnull()][column], control_97[~control_97[column].isnull()][column], equal_var = False)[1]
        output.append([column, treatment_avg, control_avg, treatment_avg - control_avg, round(p_value, 4)])
# Create dataframe and rename column names
output = pd.DataFrame(output, columns = ["Variable Name", "Average value (Treatment villages)", "Average value (Control villages)", "Difference (Treat - Control)", "p-value"])
output.sort_values(by = "Variable Name").reset_index(drop = True)

Variable Name  Average value (Treatment villages)  \
0             age                           10.716991   
1        dist_cap                          150.829074   
2        dist_sec                            2.453122   
3           fam_n                            7.281327   
4             grc                            3.531599   
5          hohage                           43.648828   
6          hohedu                            2.663139   
7          hohsex                            0.924656   
8          hohwag                          544.339544   
9           indig                            0.325986   
10       min_dist                          107.152915   
11             sc                            0.822697   
12            sex                            0.519317   
13  welfare_index                          655.428377   

    Average value (Control villages)  Difference (Treat - Control)  p-value  
0                          10.742023                     -0.025032   0.4784  
1                         153.769730                     -2.940656   0.0011  
2                           2.507662                     -0.054540   0.0427  
3                           7.302469                     -0.021142   0.4290  
4                           3.543050                     -0.011450   0.6895  
5                          44.276918                     -0.628090   0.0000  
6                           2.590348                      0.072791   0.0104  
7                           0.922947                      0.001709   0.5721  
8                         573.163558                    -28.824015   0.0003  
9                           0.332207                     -0.006222   0.2459  
10                        103.237854                      3.915060   0.0000  
11                          0.815186                      0.007511   0.0965  
12                          0.505052                      0.014265   0.0122  
13                        659.579100                     -4.150723   0.0015

    2. (4pt) Did you find any statistically significant differences between treatment and control villages?

Variables "sex", "dist_sec", "dist_cap", "hohedu", "hohwag", "hohage", "welfare_index", "min_dist" are the only ones that have statistically significant differences between treatment and control villages as they have p-values that are less than a typical alpha value (i.e. 0.05).

    3. (8pt) Why do we focus on 1997 differences only?

We want to focus on 1997 differences only because we want to analyze whether randomization was performed correctly here. The pre-treatment time, which is 1997 demographic, is observable for us to see that. If the treated group and control groups are similar, they should also be similar in other characteristics we do not observe in the data.

    4. (8pt) Why does it matter if treatment and control villages differ?

It would matter if there are differences between treatment and control villages because it violates the assumption of the two groups having similar trends before the treatment. In that case, we cannot attribute the change in output to the treatment, and no causal interpretation can be established. 

# Measuring impact (45pt)

Next, we measure the impact of Progresa. You already did it with CS and BA estimator, so the only thing to do is essentially to repeat the previous with DiD estimator. DiD relaxes the identifying assumptions but you pay the price in the form of stricter data requirements and more complex analysis.

    1. (2pt) First, let’s just compare group averages. Now you need four groups: treated and control, before and after treatment. DiD is the difference in the trends for treated and control groups. Compute these group averages and the corresponding DiD estimator.
    Hint: it should be 0.0313.

In [4]:
# Filter, groupby, and compute means for different groups
difference_data = progresa_data[progresa_data.poor == "pobre"].groupby(["year", "progresa"]).sc.mean().values
# Compute difference betweeen contol and treatment groups
difference_control = difference_data[2] - difference_data[0]
difference_treatment = difference_data[3] - difference_data[1]
# Compute the difference in difference estimator
did_estimator = difference_treatment - difference_control
# Print result statements
print("Difference in control group averages:", difference_control)
print("Difference in treatment group averages:", difference_treatment)
print("The corresponding DiD estimator is:", did_estimator)

Difference in control group averages: -0.007549046327276487
Difference in treatment group averages: 0.023782233992046597
The corresponding DiD estimator is: 0.031331280319323085


Now it’s time to introduce regression. You should regress the schooling outcome on pre/post reform indicator (i.e. year), treatment/control indicator (i.e. progresa), and their corresponding interaction effect. You should only include the treatment and control groups (i.e. poor), not “rich” families.

    2. (4pt) Estimate the effect using DiD simple regression (no other covariates).

In [5]:
# Filter for only poor data
poor_data = progresa_data[(progresa_data.poor == "pobre")]
# Perfrom simple rergression 
did_simple_regression = smf.ols("sc ~ progresa * I(year == 98)", data = poor_data).fit()
did_simple_regression.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                     sc   R-squared:                       0.001
Model:                            OLS   Adj. R-squared:                  0.001
Method:                 Least Squares   F-statistic:                     28.31
Date:                Mon, 18 Apr 2022   Prob (F-statistic):           2.76e-18
Time:                        01:20:21   Log-Likelihood:                -26242.
No. Observations:               58372   AIC:                         5.249e+04
Df Residuals:                   58368   BIC:                         5.253e+04
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
===========================================================================================================
                                              coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------------------------------
Intercept                                   0.8152      0.003    233.182      0.000       0.808       0.822
progresa[T.basal]                           0.0075      0.004      1.691      0.091      -0.001       0.016
I(year == 98)[T.True]                      -0.0075      0.005     -1.480      0.139      -0.018       0.002
progresa[T.basal]:I(year == 98)[T.True]     0.0313      0.006      4.835      0.000       0.019       0.044
==============================================================================
Omnibus:                    15346.988   Durbin-Watson:                   1.397
Prob(Omnibus):                  0.000   Jarque-Bera (JB):            30608.651
Skew:                          -1.711   Prob(JB):                         0.00
Kurtosis:                       3.937   Cond. No.                         7.67
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

    3. (4pt) Interpret all the coefficients.

Before the progresa program, the schooling rate in the control villages was about 81.5%. The schooling rate in the treated villages before treatment improves by 0.75 percentage points. The schooling rate in control villages falls by 0.75 percentage points after the progresa. However, following progresa, the education rate in treated villages increased by 3.13 percentage points.

    4. (4pt) Report the result: it should be the same as above. Is it statistically significant?

The regression model shows that the difference in the trends for the treated and control groups is 0.0313, which is the same as above. It is statistically significant because the p-value is less than a typical alpha value (i.e., 0.05).

    5. (3pt) Now estimate the effect using multiple regression–include all relevant control variables.

In [6]:
# Perfrom multi rergression 
did_multi_regression = smf.ols("sc ~ progresa * I(year == 98) + sex + fam_n + min_dist + dist_sec + hohedu+ hohwag + hohage + age", data = poor_data).fit()
did_multi_regression.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                     sc   R-squared:                       0.281
Model:                            OLS   Adj. R-squared:                  0.281
Method:                 Least Squares   F-statistic:                     2073.
Date:                Mon, 18 Apr 2022   Prob (F-statistic):               0.00
Time:                        01:20:21   Log-Likelihood:                -16644.
No. Observations:               58344   AIC:                         3.331e+04
Df Residuals:                   58332   BIC:                         3.342e+04
Df Model:                          11                                         
Covariance Type:            nonrobust                                         
===========================================================================================================
                                              coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------------------------------
Intercept                                   1.4266      0.010    144.359      0.000       1.407       1.446
progresa[T.basal]                           0.0010      0.004      0.276      0.783      -0.006       0.008
I(year == 98)[T.True]                       0.0274      0.004      6.313      0.000       0.019       0.036
progresa[T.basal]:I(year == 98)[T.True]     0.0313      0.006      5.699      0.000       0.021       0.042
sex                                         0.0331      0.003     12.405      0.000       0.028       0.038
fam_n                                      -0.0002      0.001     -0.365      0.715      -0.001       0.001
min_dist                                    0.0007   3.21e-05     21.726      0.000       0.001       0.001
dist_sec                                   -0.0098      0.001    -15.973      0.000      -0.011      -0.009
hohedu                                      0.0078      0.001     13.411      0.000       0.007       0.009
hohwag                                  -1.136e-06   1.92e-06     -0.592      0.554   -4.89e-06    2.62e-06
hohage                                      0.0003      0.000      2.287      0.022    4.25e-05       0.001
age                                        -0.0657      0.000   -143.659      0.000      -0.067      -0.065
==============================================================================
Omnibus:                     5508.809   Durbin-Watson:                   1.488
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             7238.892
Skew:                          -0.861   Prob(JB):                         0.00
Kurtosis:                       3.111   Cond. No.                     6.86e+03
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 6.86e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

    6. (3pt) Compare the results. Is the multiple-regression version similar? Is it statistically significant?
    
    
    

After comparing the result, the multiple-regression version of the coefficient is similar to the simple regression version. The interaction effect of the multiple regression is 0.0313, and the one for the simple regression is 0.0313 as well. It is statistically significant as it has a p-value less than a typical alpha value (i.e., 0.05)

    7. (5pt) What are 95% confidence intervals for this estimator? Does this encompass all other estimates you received in this and in the previous PS?
    Hint: the upper boundary of 95% CI should be 0.0409.

In [7]:
# Display the 95% confidence intervals for the multi regression 
did_multi_regression.conf_int(alpha = 0.05, cols = None)

0         1
Intercept                                1.407199  1.445937
progresa[T.basal]                       -0.006357  0.008439
I(year == 98)[T.True]                    0.018883  0.035886
progresa[T.basal]:I(year == 98)[T.True]  0.020565  0.042126
sex                                      0.027862  0.038319
fam_n                                   -0.001356  0.000931
min_dist                                 0.000633  0.000759
dist_sec                                -0.011051 -0.008635
hohedu                                   0.006675  0.008960
hohwag                                  -0.000005  0.000003
hohage                                   0.000042  0.000552
age                                     -0.066565 -0.064773

The 95% confidence interval for this estimator is 0.020565-0.042126. That means that we are 95% confident that the effect of treatment after the campaign would increase the schooling rate by 2.057 percentage points to 4.213 percentage points. This does encompass all other estimates in the previous problem set.

    8. (8pt) What is the identifying assumption behind this DiD estimator? Would you be able to test it to a certain extent using the dataset here? Explain!
    Hint: what do you expect to see when comparing different villages? The same villages over time? Do you have this information in these data?

The DiD estimator's identifying assumption is: In the absence of the treatment, the trend change over time (from 1997 to 1998) would have been similar in the treatment and the control group. From question 1.1, using the data we can observe to predict the characteristics we cannot see. As we noticed some statistically significant differences between the two groups, we should be more careful when claiming the assumption is convincing.

    9. (8pt) Compare this assumption with the assumptions behind CS and BA estimator. Which ones do you find more plausible? Why?
    Base your claims in the institutional settings: it is possibly imperfect randomized experiment in poor rural villages. Do you think some assumptions are more likely satisfied than others?

I don't think any of these assumptions are plausible. We see from question 1.1 that we didn't do great randomization for some of the variables in poor rural villages, which may lead to different trends before the treatment. We also know from the previous problem set that the cross-sectional and before-after estimators are not convincing either.

I find the before-after assumption to be satisfied more likely out of all the assumptions. This is because the likelihood of every other variable except those in the model being consistence is lower than the likelihood of variables being consistent over time.

    10. (4pt) Based on all your work you did above–what is your conclusion about the efficacy of the Progresa program?

Based on all the work done above and in the previous problem set. I can conclude that the progresa program improves the schooling rate if the assumption is true. However, the efficiency of the progresa program is not ideal as the percentage point improvement in the schooling rate is not high. If the assumption is not true, we cannot conclude the statement as other factors may lead to the increase.

# Leidner etal.(2021) (25pt)

Leidner et al. (2021) (available on canvas in files/readings) analyse the effect of online versus in person instruction from COVID-19 perspective; they find that online instructions help to avoid a substantial number of cases.

    1. Read the paper.
    
    2. (3pt) Let us focus on the unmatched analysis. What kind of counties are the authors comparing in the paper? How many counties do they have? How many of these had remote, how many in-person instructions?

In the unmatched analysis, they are comparing university counties with non-university counties. Within university counties, they are looking at remote vs. in-person instruction. There are 133 counties with large colleges and universities, the 101 in which classes started from July 27 to August 28 were included in the analysis. There are also 3,009 non-university counties. Among the 101 university counties, 22 had remote, and 79 had in-person.

    3. (2pt) Which time period are they looking at?

The time period they are looking at is the 21 days before and 21 days after the start of the classes.

    4. (4pt) What is treatment in this case? (You can define it in different ways).

The treatment variable is in-person instruction at university counties. The control group is the remote instruction university counties, and the treatment group is in-person instruction at university counties.

    5. (3pt) What is the main outcome measure the authors discuss?

The main outcome measure the authors discuss is the average county COVID-19 incidence in university counties.

    6. (4pt) Why do authors analyze percentage positive results in testing?

The authors analyze percentage positive results in testing because they want to make sure that the county COVID-19 incidence is not confounded by different testing rates. For example, if a university county with in-person attendance is tested way more than a remote instruction, the COVID-19 incidence would be higher only because of the number of tests done to the in-person instruction county. 

    7. (4pt) The authors provide the figures for COVID-19 incidence (in table). Take the numbers and use those to do DiD youself! Show the calculations and the answer.
    Note: You don’t really need computers, the table provides the four averages, so you only have to compute the differences of those.
    Hint: effect should be 11.8 (or -11.8, depending which way you define it).

In [8]:
# Compute the difference in difference estimator for the article
leinder_did_estimator = (23.9 - 15.3) - (14.7 - 17.9)
print("The difference in difference estimator is:", leinder_did_estimator)

The difference in difference estimator is: 11.799999999999997


    8. (5pt) What are the identifying assumptions? Do you find these credible?
    Hint: read about the limitations of the study and think which limitations are about identification.

The identifying assumption for this study is that COVID-19 incidence rates in counties would have been similar for both treatment and control without the treatment being applied. From reading the limitations part of the study, it is realized that this study did not adjust for mitigation strategies (e.g., mask and social distancing requirements and limits on large crowds and athletic events) implemented at local or state levels or at colleges and universities, which could have affected the association between the institution’s opening and county-level incidence.  

The graph in the article shows the trend before and after the treatment. We can tell that before treatment, university counties with in-person instructions have a similar trend with remote instructions. Therefore, I find these to be credible.

# Finally How much time spent on this PS?

I spent about 10 hours on this PS.